### PR TITLE
control-service: secrets service unit tests

### PR DESCRIPTION
**Pull request by dakodakov**
Add unit tests for the secrets service.
Minor code changes to address redundancies, feedback from previous review.

### CODE REVIEW

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java [hunk 1]
```diff
@@ -23,6 +27,9 @@
 @ConditionalOnProperty(value = "featureflag.vault.integration.enabled")
 public class VaultJobSecretsService implements com.vmware.taurus.secrets.service.JobSecretsService {
 
+  @Value("${vdk.vault.size.limit}")
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:30]: I forgot to mention the last time but why `vdk.` prefix when all other variables use `datajobs.` prefix. 

Technically `vdk.` is better one since this is the product name. 

`datajobs` is more of legacy prefix since the time the control service was called only Data Jobs Service. 

Still would users wonder about the difference ?

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java [hunk 2]
```diff
@@ -23,6 +27,9 @@
...
+  private int sizeLimit = 1048576;
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:31]: Please, always include the unit in a variable name - in this case I think it's `bytes` , right ? 
E.g "sizeLimitBytes"  and `vdk.vault.size.limit.bytes` or something along those lines. 

Future maintainer and administrators configuring it would be very grateful for this clarity.

Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:31]: The purist in me must mention : Avoid "magic numbers": The number 1048576 is hard-coded into the sizeLimit variable. Better it would be constant VAULT_SIZE_LIMIT_DEFAULT ...

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java [hunk 3]
```diff
@@ -57,7 +64,17 @@ public void updateJobSecrets(String jobName, Map<String, Object> secrets) {
...
 
-    vaultJobSecrets.setSecretsJson(new JSONObject(updatedSecrets).toString());
+    String updatedSecretsString = new JSONObject(updatedSecrets).toString();
+    try {
+      if (updatedSecretsString.getBytes("UTF-8").length > sizeLimit) {
+        throw new DataJobSecretsSizeLimitException(
+            jobName, "Secret size exceeds configured limit of:" + sizeLimit);
+      }
+    } catch (UnsupportedEncodingException e) {
+      throw new DataJobSecretsException(jobName, e.getMessage());
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:74]: nit: e.toString() might be more helpful as it include the class name.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java [hunk 4]
```diff
@@ -57,7 +64,17 @@ public void updateJobSecrets(String jobName, Map<String, Object> secrets) {
                       return entry.getValue();
                     }));
 
-    vaultJobSecrets.setSecretsJson(new JSONObject(updatedSecrets).toString());
+    String updatedSecretsString = new JSONObject(updatedSecrets).toString();
+    try {
+      if (updatedSecretsString.getBytes("UTF-8").length > sizeLimit) {
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java:69]: Again magic word. Fortunatelh since java 8 there is [StandardCharsets.UTF_8](https://docs.oracle.com/javase/8/docs/api/java/nio/charset/StandardCharsets.html#UTF_8)

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/secrets/DataJobsSecretsControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+@ExtendWith(MockitoExtension.class)
+class DataJobsSecretsControllerTest {
+
+  @Mock private JobSecretsService jobSecretsService;
+
+  @InjectMocks private DataJobsSecretsController controller;
+  @InjectMocks private NoOpDataJobsSecretsController noOpsController;
+
+  @Test
+  void testDataJobSecretsUpdate() {
```
Comment (tozka) [inline projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/secrets/DataJobsSecretsControllerTest.java:42]: Have you thought of those edge cases 
- Can job name be null ? (if yes we need a test)
- team name be null ? 
- Can the secrets map be null or empty ?
- Can the secrets map contain invalid characters  ?

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java (hunk 1 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed to be consistent with old properties

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java (hunk 2 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed.

Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java (hunk 3 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed.

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java (hunk 4 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/service/vault/VaultJobSecretsService.java]: fixed.

File: projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/secrets/DataJobsSecretsControllerTest.java (hunk 5 above)
Comment (dakodakov) [reply projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/secrets/DataJobsSecretsControllerTest.java:42]: addressed.

**Review by tozka — APPROVED** [review]

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/controller/DataJobsSecretsController.java [hunk 6]
```diff
@@ -30,14 +29,11 @@
 public class DataJobsSecretsController implements DataJobsSecretsApi {
   static Logger log = LoggerFactory.getLogger(DataJobsSecretsController.class);
 
-  private final FeatureFlags featureFlags;
-
   private final JobSecretsService secretsService;
 
   @Autowired
   public DataJobsSecretsController(
       FeatureFlags featureFlags, @Nullable JobSecretsService secretsService) {
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/controller/DataJobsSecretsController.java:36]:
```suggestion
      @Nullable JobSecretsService secretsService) {
```

File: projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/controller/DataJobsSecretsController.java [hunk 7]
```diff
@@ -9,7 +9,6 @@
 import com.vmware.taurus.base.FeatureFlags;
```
Comment (mivanov1988) [inline projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/secrets/controller/DataJobsSecretsController.java:9]:
```suggestion
```
